### PR TITLE
storage: add the connection id to the kafka group id

### PR DESCRIPTION
**Pull request by guswynn**
As discussed on slack. One of the final (non-QA) changes required by https://github.com/MaterializeInc/materialize/issues/13534

cc @morsapaes this will need to be documented when we document the kafka commit behavior

### Motivation

  * This PR adds a known-desirable feature.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

### CODE REVIEW

File: src/storage/src/source/kafka.rs [hunk 1]
```diff
@@ -646,10 +652,12 @@ async fn create_kafka_config(
     kafka_config.set(
         "group.id",
         &format!(
-            "{}materialize-{}-{}",
+            "{}materialize-{}-{}-{}-source-{}",
             group_id_prefix.unwrap_or_else(String::new),
             environment_id,
-            name
+            connection_name,
```
[the remark this answers is not in the document]
Comment (benesch) [reply src/storage/src/source/kafka.rs:658]: Yeah, can we just do `{prefix}materialize-{environment_id}-{connection_id}-{source_id}`? It's by definition a source, too!

[the remark this answers is not in the document]
Comment (guswynn) [reply src/storage/src/source/kafka.rs:658]: sounds good, ill adjust

[the remark this answers is not in the document]
Comment (guswynn) [reply src/storage/src/source/kafka.rs]: pushed with this change!

**Review by benesch — APPROVED** [review]
I pushed up two additional commits. Details in commit message, but in short:

  * The first commit adds a new `set-from-sql` testdrive action that sets a variable to the value of a SQL query.
  * The second commit uses this new functionality to additionally test that the `GROUP ID PREFIX` Kafka source option works as expected.

**Comment by benesch** [comment]
Going to merge this so that I can put up a PR with some other testdrive changes based on top of this one. Let me know if you have any complaints about my commits post merge and I will address!

File: test/testdrive/kafka-commit.td [hunk 2]
```diff
@@ -19,20 +21,52 @@ three
...
+one
+two
+three
+
+$ set-from-sql var=consumer-group-id
+SELECT
+  'materialize-' || mz_environment_id() || '-' || c.id || '-' || s.id
+FROM mz_sources s
+JOIN mz_connections c ON c.id = s.connection_id
+WHERE s.name = 'topic'
```
Comment (guswynn) [inline test/testdrive/kafka-commit.td:43]: til about `||` in sql

**Comment by guswynn** [comment]
thanks @benesch !

**Pull request merged**